widget_driver/lib/src/drivable_widget.dart

```diff
@@ -55,12 +57,32 @@ class _DriverWidgetState<Driver extends WidgetDriver> extends State<DrivableWidg
     return widget.build(context);
   }
 
+  @override
+  void didUpdateWidget(covariant DrivableWidget<Driver> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_driverExists) {
+      widget.driverProvider.updateDriverProvidedProperties(_driver!);
+    }
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (_driverExists) {
+      // If didChangeDependencies get's called this means your driver depends on a value from the BuildContext,
+      // which the flutter framework marked as "has changed".
+      // Thus we need to rebuild the driver to retrieve the latest value.
+      _markDriverToBeRebuilt();
+    }
+  }
+
   Driver _getDriverAndSetupUpIfNeeded(BuildContext context) {
-    // The driver was already created. Nothing more to do. Just return it.
-    if (_driver != null) {
+    if (_driverExists && !_shouldRebuildDriver) {
       return _driver!;
     }
 
+    _disposeDriver();
```

Comment:
I hope you get a failing test here..
Cause now you are sort of breaking/changing logic..
Now for every build you dispose the driver.. hmm.. Probably not what we want.. 
If your tests did not break.. Then we need a new test also which checks that we do not call dispose on driver at wrong time..

Reply:
Not really.
Ok, so when we ***have a driver and we don't want to rebuild it*** we return, thereby not reaching this line.
So we are left with:
* ***we have a driver and want to rebuild it***: Great we dispose the driver and build a new one
* ***we don't have driver and we don't want to rebuild it***: Well we should probably have one, so we do want to create one. Disposing doesn't hurt, we really only delete the driver if it actually exists (see Dominiks comment)
* ***we don't have a driver and do want to rebuild it***: Well we can't really rebuild it, but we can create one... Again disposing doesn't really do anything, because the driver doesn't exist.

I hope my logic isn't broken 😄

Reply:
aha, yeah. You are right. sorry about that.